aiida_quantumespresso/workflows/pw/base.py

```diff
@@ -195,14 +195,18 @@ def get_builder_from_protocol(
             parameters['SYSTEM']['nspin'] = 2
             parameters['SYSTEM']['starting_magnetization'] = starting_magnetization
 
-        builder.pw['code'] = code  # pylint: disable=no-member
-        builder.pw['pseudos'] = pseudo_family.get_pseudos(structure=structure)  # pylint: disable=no-member
-        builder.pw['structure'] = structure  # pylint: disable=no-member
-        builder.pw['parameters'] = orm.Dict(dict=parameters)  # pylint: disable=no-member
-        builder.pw['metadata'] = inputs['metadata']  # pylint: disable=no-member
+        # pylint: disable=no-member
+        builder.pw['code'] = code
+        builder.pw['pseudos'] = pseudo_family.get_pseudos(structure=structure)
+        builder.pw['structure'] = structure
+        builder.pw['parameters'] = orm.Dict(dict=parameters)
+        builder.pw['metadata'] = inputs['pw']['metadata']
+        if 'parallelization' in inputs['pw']:
+            builder.pw['parallelization'] = orm.Dict(dict=inputs['pw']['parallelization'])
```

Comment:
can't the parallelization already be an instance of `Dict`?

Reply:
Usually I add the overrides as one big regular ol' dictionary, and as far as I know the loaded defaults are also just in a dictionary format? 

Second, I was actually also sort of hoping that the `Dict` constructor would be idempotent, but alas:

```Python
In [3]: d = Dict(dict={'a': 1})

In [4]: d2 = Dict(dict=d)
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-4-b2188aaaaf40> in <module>
----> 1 d2 = Dict(dict=d)

~/envs/aiida-dev/code/aiida-core/aiida/orm/nodes/data/dict.py in __init__(self, **kwargs)
     60         super().__init__(**kwargs)
     61         if dictionary:
---> 62             self.set_dict(dictionary)
     63 
     64     def __getitem__(self, key):

~/envs/aiida-dev/code/aiida-core/aiida/orm/nodes/data/dict.py in set_dict(self, dictionary)
     81             # Clear existing attributes and set the new dictionary
     82             self.clear_attributes()
---> 83             self.update_dict(dictionary)
     84         except exceptions.ModificationNotAllowed:  # pylint: disable=try-except-raise
     85             # I reraise here to avoid to go in the generic 'except' below that would raise the same exception again

~/envs/aiida-dev/code/aiida-core/aiida/orm/nodes/data/dict.py in update_dict(self, dictionary)
     98         :param dictionary: a dictionary with the keys to substitute
     99         """
--> 100         for key, value in dictionary.items():
    101             self.set_attribute(key, value)
    102 

AttributeError: 'Dict' object has no attribute 'items'
```

I think we should be working on making the `Dict` more like `dict`, so this is no longer an issue.

Reply:
That might not be as straighforward as you imagine as far as the expected behavior is concerned. For example, consider the behavior of `dict`:
```
In [1]: d = dict(a=1)

In [2]: id(d)
Out[2]: 140302121245760

In [3]: e = dict(d)

In [4]: id(e)
Out[4]: 140302119318464
```
if you apply this behavior to `Dict` and then use it in the way you have, you would be creating a new `Dict` node. I would argue that this is not what you want here. If it is already a `Dict` node, you want to use exactly that node that was an input and not create a new one. You create a new one only when you really have to

Reply:
Fair point. 👍  I was thinking the initial `Dict` node might not have been stored in the database, but actually, it could very well be; and this might be exactly the reason to support providing `Dict` nodes as an input in the `overrides`.

The same point could be made for other inputs, however. Consider for example the `clean_workdir` input for the `PwBaseWorkChain`:

```Python
In [1]: PwBaseWorkChain = WorkflowFactory('quantumespresso.pw.base')

In [2]: code = load_code(53)
   ...: structure = load_node(54)
   ...: 
   ...: clean_workdir = Bool(False)
   ...: clean_workdir.store()
Out[2]: <Bool: uuid: 26c7022a-6e63-49ab-9421-52bf016da014 (pk: 2384) value: False>

In [3]: overrides = {
   ...:     'clean_workdir': clean_workdir
   ...: }

In [4]: builder = PwBaseWorkChain.get_builder_from_protocol(
   ...:     code=code,
   ...:     structure=structure,
   ...:     overrides=overrides
   ...: )

In [5]: builder.clean_workdir
Out[5]: <Bool: uuid: f07c9464-37d6-459c-ba67-d560ce58b4cc (unstored) value: False>
```

So we'd have to update the logic in the `get_builder_from_protocol()` method everywhere, or simply only allow `isinstance(overrides, dict)`. Considering the fact that there might be a use case for re-using nodes, I'd suggest the former, but I will do this in a separate PR with a proper explanation.

However, I still think we should promote `Dict` to a proper `BaseType`, and make sure the behaviour is as close to `dict` as possible. But I suggest we move that discussion to an issue I had opened that discusses similar changes to `List` and `Dict`:

https://github.com/aiidateam/aiida-core/issues/4495

I'll add a note re this discussion there.